packages/select/src/components/select/suggest.tsx

```diff
@@ -68,6 +74,7 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
 
     public state: ISuggestState<T> = {
         isOpen: (this.props.popoverProps && this.props.popoverProps.isOpen) || false,
+        selectedItem: this.props.selectedItem
```

Comment:
what you've implemented here is `defaultSelectedItem` - the initial uncontrolled state.

`selectedItem` should _control_ the state field by updating it when props change: if i pass a new `selectedItem` prop value, that should be reflected in later renders. this will not happen with your current code.

Reply:
this is analogous to `value`, `defaultValue` and `onChange` on an `<input>` tag. 
https://reactjs.org/docs/uncontrolled-components.html

Reply:
You're right, I made the change to fix one bug that I encountered on my code but I missed that once the `selectedItem` has been initialised, the later changes won't do anything at all. I'll make a change right away.

Reply:
@giladgray In controlled mode, do you think the Suggest should still set the `selectedItem` state in the `handleItemSelect` handler? The `setState` will be called anyway because of the `nextOpenState` change, otherwise we could save a render cycle.